packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -91,14 +96,34 @@ export class EthersEth1Notifier extends (EventEmitter as { new(): Eth1EventEmitt
       `Last processed block number: ${this.lastProcessedEth1BlockNumber}, ` +
       `last processed deposit index: ${this.lastProcessedDepositIndex}`
     );
+  }
+
+  public getEth1DataSource(): AsyncIterable<{timestamp: number; eth1Data: Eth1Data}> {
+    return this.eth1DataSource;
+  }
+
+  public async startScanEth1DataForGenesis(): Promise<void> {
     const headBlockNumber = await this.provider.getBlockNumber();
-    // process historical unprocessed blocks up to curent head
-    // then start listening for incoming blocks
-    this.processBlocks(headBlockNumber).then(() => {
-      if(this.started) {
-        this.provider.on("block", this.onNewEth1Block.bind(this));
-      }
-    });
+    this.isScanEth1ForGenesis = true;
+    this.logger.info(
+      `Start scanning Eth1Data last block =${this.lastProcessedEth1BlockNumber} headBlockNumber=${headBlockNumber}`
+    );
+    while (this.lastProcessedEth1BlockNumber < headBlockNumber && this.isScanEth1ForGenesis) {
```

Comment:
This is only working because schlesi had minimal deposits within 4 mins after contract deployment so genesis is formed after couple of blocks 
Consider following scenario which will happen in mainnet and maybe multiclient testnet.
1. deposit contract is deployed at block n (I made that number up)
2. We will wait a month for minimal deposits and genesis time (31day*24hours * 4 blocks per hour ~= 3000 blocks)
3. We take usually around 1s to fetch and parse eth1 block (in which case this loop is gonna take 1 hour to execute this loop)

When mainnet launches it will probably be more than 1 month so we would take couple hours just to find genesis

Reply:
If you look into `processDepositEvents` you'll see I don't call `getBlock()` for all blocks. I only call it until we have enough `MIN_GENESIS_ACTIVE_VALIDATOR_COUNT` so I don't think we have issue with that scenario because we'll just keep filtering deposit events and parse them.

Reply:
Oh, you are going exactly what I'm saying:
onNewEth1Block ->processBlocksInBulk (deployedAt, eth1_head - follow distance)  -> getDepositEvents

processBlocksInBulk should be called processDepositEvents as it's not processing blocks just deposit events.

Also this needs a lot of renaming like onNewEth1Block is called to process bunch of blocks etc. Also, handle cases from https://github.com/ChainSafe/lodestar/issues/889#issuecomment-626302180